tests/fcgi-auth.c

```diff
@@ -21,7 +21,7 @@ int main (void) {
 		if (((p = getenv("QUERY_STRING")) == NULL) ||
 		    (strcmp(p, "ok") != 0 && strcmp(p, "var") != 0)) {
 			printf("Status: 403 Forbidden\r\n\r\n");
-		} else if (((p = getenv("QUERY_STRING")) != NULL) || strcmp(p, "var") == 0) {
+		} else if (((p = getenv("QUERY_STRING")) != NULL) && strcmp(p, "var") == 0) {
```

Comment:
No need to repeat (p = getenv("QUERY_STRING")) here.  It was done above.  Still need to check (p != NULL).  Sorry that I hadn't mentioned this in prior comment.  (BTW, feel free to rebase and force-push changes to this dev branch.)

I'd move the char* p into the loop, too, and remote the unnecessary Content-Type response header (not required for FastCGI authorizer).  (Untested code typed directly into web browser:)
<pre>
                /* wait for fastcgi authorizer request */

                /* Status: 200 OK to allow access is implied
                 * if Status header is not included in response */
                char *p = getenv("QUERY_STRING");
                if (p != NULL && 0 == strcmp(p, "ok")) {
                        printf("\r\n");
                } else if (p != NULL && 0 == strcmp(p, "var")) {
                        printf("Variable-X-LIGHTTPD-FCGI-AUTH: LighttpdTestContent\r\n\r\n");
                } else {
                        printf("Status: 403 Forbidden\r\n\r\n");
                }
</pre>
I also renamed LIGHTYTEST to X-LIGHTTPD-FCGI-AUTH, though doing so here will mean updating other tests in this pull request which looking for that header.